tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers_test.ts

```diff
@@ -156,6 +239,206 @@ describe('Debugger reducers', () => {
+      expect(nextState.alerts.alertsLoaded.lastLoadedTimeInMs).toBeGreaterThan(
+        0
+      );
+      expect(nextState.alerts.numAlerts).toBe(2);
+      expect(nextState.alerts.alertsBreakdown).toEqual({
+        [AlertType.INF_NAN_ALERT]: 2,
+      });
+      expect(Object.keys(nextState.alerts.alerts)).toEqual([
+        AlertType.INF_NAN_ALERT,
+      ]);
+      const alertsOfType = nextState.alerts.alerts[AlertType.INF_NAN_ALERT];
+      expect(Object.keys(alertsOfType).length).toBe(2);
+      expect(alertsOfType[0]).toEqual(alert0);
+      expect(alertsOfType[1]).toEqual(alert1);
+      expect(Object.keys(nextState.alerts.executionIndices)).toEqual([
+        AlertType.INF_NAN_ALERT,
+      ]);
+      const executionIndices =
+        nextState.alerts.executionIndices[AlertType.INF_NAN_ALERT];
+      expect(executionIndices[1]).toBe(11);
+    });
+
+    it('Updates alerts data: existing alert types other than the loaded', () => {
+      const alert0 = createTestInfNanAlert({
+        op_type: 'RealDiv',
+        execution_index: 10,
+      });
+      const alert1 = createTestInfNanAlert({
+        op_type: 'Log',
+        execution_index: 11,
+      });
+      const tensorShapeAlert: Alert = {
+        alert_type: AlertType.TENSOR_SHAPE_ALERT,
+      };
+      const state = createDebuggerState({
+        activeRunId: '__default_debugger_run__',
+        alerts: createAlertsState({
+          alertsLoaded: {
+            state: DataLoadState.LOADING,
+            lastLoadedTimeInMs: null,
+          },
+          numAlerts: 2,
+          alertsBreakdown: {
+            [AlertType.INF_NAN_ALERT]: 1,
+            [AlertType.TENSOR_SHAPE_ALERT]: 1,
+          },
+          alerts: {
+            [AlertType.INF_NAN_ALERT]: {0: alert0},
+            [AlertType.TENSOR_SHAPE_ALERT]: {0: tensorShapeAlert},
+          },
+        }),
+      });
+
+      const nextState = reducers(
+        state,
+        actions.alertsOfTypeLoaded({
+          numAlerts: 3,
+          alertsBreakdown: {
+            [AlertType.INF_NAN_ALERT]: 2,
+            [AlertType.TENSOR_SHAPE_ALERT]: 1,
+          },
+          begin: 1,
+          end: 2,
+          alertType: 'InfNanAlert',
+          alerts: [alert1],
+        })
+      );
+      expect(nextState.alerts.numAlerts).toBe(3);
+      expect(nextState.alerts.alertsBreakdown).toEqual({
+        [AlertType.INF_NAN_ALERT]: 2,
+        [AlertType.TENSOR_SHAPE_ALERT]: 1,
+      });
+      expect(Object.keys(nextState.alerts.alerts).length).toBe(2);
+      const infNanAlerts = nextState.alerts.alerts[AlertType.INF_NAN_ALERT];
+      expect(Object.keys(infNanAlerts).length).toBe(2);
+      expect(infNanAlerts[0]).toEqual(alert0);
+      expect(infNanAlerts[1]).toEqual(alert1);
+      const tensorShapeAlerts =
+        nextState.alerts.alerts[AlertType.TENSOR_SHAPE_ALERT];
+      expect(Object.keys(tensorShapeAlerts).length).toBe(1);
+      expect(tensorShapeAlerts[0]).toEqual(tensorShapeAlert);
```

Comment:
Here and elsewhere: do you prefer this way of assertion because you don't care how they are keyed? Otherwise, did you consider writing,

```ts
expect(nextState.alert.alerts).toEqual({
  [AlertType.INF_NAN_ALERT]: {
    'somekey0': alert0,
    'somekey'1: alert1,
  },
  [AlertType.TENSOR_SHAPE_ALERT]: {
    'somekey0': tensorShapeAlert,
  },
})
```

Reply:
I slightly prefer this way, because it is a little more granular and when mismatch happens, we can quickly identify the line responsible for the failure.